sympy/matrices/matrices.py

```diff
@@ -2561,11 +2561,19 @@ def diagonal_solve(self, rhs):
             raise TypeError("Size mis-match")
         return self._diagonal_solve(rhs)
 
-    def dot(self, b):
+    def dot(self, b, **kwargs):
         """Return the dot product of two vectors of equal length. ``self`` must
         be a ``Matrix`` of size 1 x n or n x 1, and ``b`` must be either a
         matrix of size 1 x n, n x 1, or a list/tuple of length n. A scalar is returned.
 
+        In case one or both of the vector is complex then we need to calculate the inner product
+        of the two vectors. To calculate the inner product we call the inner_product function.
+
+        The expected kwargs are hermitian and conjugate_convention. Hermitian takes two True or False
+        depending on wether dot product is to be calculated or inner product is to be used.
+        conjugate_convention has basically two inputs ("maths", "left", "math") or ("physics", "right").
```

Comment:
This isn't clear that the input should be one of these. Also there should be an example showing how to use the argument.